### PR TITLE
[RISCV][TTI] Implement cost of intrinsic active_lane_mask

### PR DESCRIPTION
**Pull request by arcbbb**
This patch uses the argument type to infer the LMUL cost for the index generation, add, and comparison.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp [hunk 1]
```diff
@@ -863,6 +863,25 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     }
     break;
   }
+  case Intrinsic::get_active_lane_mask: {
+    if (ST->hasVInstructions()) {
+      Type *ExpRetTy = VectorType::get(
+          ICA.getArgTypes()[0], cast<VectorType>(RetTy)->getElementCount());
+      auto LT = getTypeLegalizationCost(ExpRetTy);
+
+      // vid.v   v8
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:872]: `vid.v v8` will always be hoisted I think, should we take this into consideration?

Comment (arcbbb) [reply llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:872]: I thought for that case we may need an argument to know if the user want to include `vid.v` in the cost.
However upon reviewing BasicTTIImpl.h, I notice that the cost is calculated as a sum of  `uadd_sat` and `ICmp`.
I think I can follow that approach.

File: llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp [hunk 2]
```diff
@@ -863,6 +863,21 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     }
     break;
   }
+  case Intrinsic::get_active_lane_mask: {
+    if (ST->hasVInstructions()) {
+      Type *ExpRetTy = VectorType::get(
+          ICA.getArgTypes()[0], cast<VectorType>(RetTy)->getElementCount());
+      auto LT = getTypeLegalizationCost(ExpRetTy);
+
+      // vid.v   v8  // considered hoisted
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:872]: Don't we need to count it for code size?

Comment (arcbbb) [reply llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:872]: This poses a dilemma as both statements hold true:
1) the `vid.v` would be hoisted.
2) BasicImpl doesn't count it for all CostKind.

so I am leaning toward to consistency with the base implementation.

Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:872]: ok

**Review by topperc — APPROVED** [review]
LGTM

**Review by preames — APPROVED** [review]

**Pull request merged**